httpbakery/form/form.go

```diff
@@ -0,0 +1,151 @@
+// Package form enables interactive login without using a web browser.
+package form
+
+import (
+	"net/http"
+	"net/url"
+
+	"github.com/juju/httprequest"
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/environschema.v1"
+
+	"gopkg.in/macaroon-bakery.v1/httpbakery"
+)
+
+/*
+PROTOCOL
+
+A form login works as follows:
+
+	   Client                            Login Service
+	      |                                    |
+	      | GET visitURL with                  |
+	      | "Accept: application/json"         |
+	      |----------------------------------->|
+	      |                                    |
+	      |   Login Methods (including "form") |
+	      |<-----------------------------------|
+	      |                                    |
+	      | GET "form" URL                     |
+	      |----------------------------------->|
+	      |                                    |
+	      |                  Schema definition |
+	      |<-----------------------------------|
+	      |                                    |
+	+-------------+                            |
+	|   Client    |                            |
+	| Interaction |                            |
+	+-------------+                            |
+	      |                                    |
+	      | POST data to "form" URL            |
+	      |----------------------------------->|
+	      |                                    |
+	      |                Form login response |
+	      |<-----------------------------------|
+	      |                                    |
+
+The schema is provided as a environschema.Fileds object. It is the
+client's responsibility to interpret the schema and present it to the
+user.
+*/
+
+// Objects implementing the FormProcessor interface interpret schema and
```

Comment:
s/FormProcessor/Processor/

I still think that "Get" (which is essentially the contract of this function) is
more straightforward than the uncertainly-worded "Process".

Hmm, thinking further into form analogies, how about Filler ?

    // Filler represents an object that can fill out a form. The given schema
    // represents the form template. The returned value should be compatible
    // with this.

We don't really need to talk about user interaction here - that might or might
not happen, depending on the client.